controllers/front/callbacknotification.php

```diff
@@ -238,4 +253,62 @@ public function postProcess()
         unlockCallback($lockFileName, $lockFileHandle);
         exit($message);
     }
+
+    public function processChildOrder($cart, $response, $transactionStatus, $auth_statuses, $captured_statuses, $order_state, $lockFileName, $lockFileHandle)
+    {
+        $shopOrderId = $response->shopOrderId;
+        // Check if an order exist
+        $order = getChildOrderFromUniqueId($shopOrderId);
+        if (!Validate::isLoadedObject($order)) {
+            $order_id = Order::getOrderByCartId((int) ($cart->id));
+            $orderDetail = new Order((int) $order_id);
+
+            createAltapayOrder($response, $orderDetail, 'succeeded', true);
+
+            if (!empty($response->Transactions[0]->ReconciliationIdentifiers)) {
+                $reconciliation_identifier = $response->Transactions[0]->ReconciliationIdentifiers[0]->Id;
+                $reconciliation_type = $response->Transactions[0]->ReconciliationIdentifiers[0]->Type;
+
+                saveOrderReconciliationIdentifierIfNotExists($orderDetail->id, $reconciliation_identifier, $reconciliation_type, $shopOrderId);
+            }
+            unlockCallback($lockFileName, $lockFileHandle);
+        } elseif (Validate::isLoadedObject($order)) {
+            if (in_array($transactionStatus, $auth_statuses, true) or in_array($transactionStatus, $captured_statuses, true)) {
+                setOrderStateIfNotExistInHistory($order, $order_state);
+                // Update payment status to 'succeeded'
+                $sql = 'UPDATE `' . _DB_PREFIX_ . 'altapay_child_order` 
+                    SET `paymentStatus` = \'succeeded\' WHERE `unique_id` = ' . $shopOrderId;
+                Db::getInstance()->Execute($sql);
+                $payment = $order->getOrderPaymentCollection();
+                if (isset($payment[0])) {
+                    $payment[0]->transaction_id = pSQL($shopOrderId);
+                    $payment[0]->save();
+                }
+
+                if (!empty($response->Transactions[0]->ReconciliationIdentifiers)) {
+                    $reconciliation_identifier = $response->Transactions[0]->ReconciliationIdentifiers[0]->Id;
+                    $reconciliation_type = $response->Transactions[0]->ReconciliationIdentifiers[0]->Type;
+
+                    saveOrderReconciliationIdentifierIfNotExists($order->id, $reconciliation_identifier, $reconciliation_type, $shopOrderId);
+                }
+                unlockCallback($lockFileName, $lockFileHandle);
+                exit('Order status updated to Accepted');
+            }
+        } elseif ($transactionStatus === 'epayment_declined') {
+            // Update payment status to 'declined'
+            $sql = 'UPDATE `' . _DB_PREFIX_ . 'altapay_child_order` 
+                        SET `paymentStatus` = \'declined\' WHERE `unique_id` = ' . $shopOrderId;
```

Comment:
This is potential for SQL injection. use the pSQL method.

Reply:
Escaped the variables using pSQL ✅